better-monadic-for/src/main/scala/com/olegpy/bm4/ImplicitPatterns.scala

```diff
@@ -102,19 +102,43 @@ trait ImplicitPatterns extends TreeUtils { self =>
 
   object HasImplicitPattern {
     def unapply(arg: Tree): Boolean = arg.exists {
-      case q"implicit0(${Bind(t: TermName, Typed(Ident(termNames.WILDCARD), _))})" if t != termNames.WILDCARD =>
+      case t@q"implicit0(${WildcardIdentifier(typed)})" =>
+        val andType = if(typed) " and type" else ""
+
+        reporter.error(
+          arg.pos,
+          s"implicit pattern requires an identifier, but a wildcard was used: `$t`. " +
+            s"This doesn't introduce anything into the implicit scope. You might want to remove the implicit0 pattern$andType."
+        )
+        false
+
+      case q"implicit0(${Bind(t: TermName, Typed(Ident(termNames.WILDCARD), _))})" =>
```

Comment:
Do you know why this `Ident(termNames.WILDCARD)` pattern is there? Some synthetic stuff made up earlier by the compiler, I guess?

btw. is there some tool for quickly getting a structure (similar to these patterns) from a compiler tree? All I relied on was `toString` and `summaryString`...

Reply:
IIRC the desugared trees are always transformed in this way:
`case a` => `case a @ _`
`case a: Thing` => `case a @ (_: Thing)`

There should be `showRaw` on the reflection universe which would do exactly what you're asking. It works even with runtime universe (so you can try it out in repl/worksheet) but the output is horrendous to read 😅